content/booklets/adacore-technologies-for-airborne-software/analysis.rst

```diff
@@ -0,0 +1,3180 @@
+.. index:: GNAT Static Analysis Suite (GNAT SAS)
+.. index:: GNATcheck
+.. index:: GNATstack
+.. index:: GNAT Dynamic Analysis Suite (GNAT DAS)
+.. index:: GNATtest
+.. index:: TGen
+.. index:: GNATcoverage
+.. index:: GNATfuzz
+
+Language development environments provide the foundation for AdaCore's toolchains, including support for Ada, C, C++, and Rust.
+Complementary tools support several verification activities for Ada:
+
+* GNATstack for stack checking (which also supports C),
+* the GNAT Static Analysis Suite, or GNAT SAS, for defect and vulnerability detection, code standard checking (GNATcheck),
+  and code metrics generation (GNATmetric)
+* the GNAT Dynamic Analysis Suite, or GNAT DAS, for testing (GNATtest, TGen), structural code
+  coverage analysis (GNATcoverage, which also supports C and C++), processor emulation (GNATemulator), and fuzzing (GNATfuzz).
+
+To show how AdaCore tools can be used in connection with the software life cycle processes for a system that is to be assessed
+against |do-178c|, several possible scenarios will be described:
+
+.. index:: Use Case 1a: Traditional development process excluding OOT
+.. index:: Use Case 1b: Traditional development process including OOT
+
+* Use Case 1: Traditional development process, excluding or including OOT
+   The development process produces requirements specified in text (natural language) that are implemented in Ada source code.
+   A code standard defines a set of restrictions, which may or may not include limitations on object-oriented features. Both cases need to be considered:
+
+   * Use Case 1a: No use is made of object oriented technology or related techniques
+   * Use Case 1b: Ada's OOT features are used, and the guidance in |do-332| is considered
+
+.. index:: DO-333/ED-216: Formal Methods
+.. index:: Use Case 2: Using SPARK and Formal Methods
+
+* Use Case 2: Using SPARK and Formal Methods
+   The development uses a formal description of the low-level requirements, namely SPARK / Ada 2012 contracts.
+   A formal analysis is performed, and credit is claimed on reducing the testing. The certification effort follows
+   the additional guidance from the Formal Methods Supplement (|do-333|).
+
+In the tables that appear in this chapter, the references shown in parentheses for the objectives identify the table, objective number,
+and paragraph reference for the objective in the |do-178c| standard or the relevant technology supplement. For example,
+A-2[6]: 5.3.1.a refers to Table A-2, Objective 6, paragraph 5.3.1a.
+
+.. index:: single: Use Case 1a: Traditional development process excluding OOT; Coding with Ada 2012
+
+Use case #1a: Coding with Ada 2012
+----------------------------------
+The adoption of Ada as the coding language brings a number of benefits during design, coding, and testing, both from language
+features (as summarized in the table below) and from the AdaCore ecosystem.
+
+.. index:: single: Ada language; Benefits
+
+.. _Benefits_of_the_Ada_language:
```

Comment:
We have to use a prefix for these anchors because of the risk of clashing with anchors from other courses. Also, we've been using capital letters for every word. Taking all of this into account, in this specific case, we could replace `Benefits_of_the_Ada_language` by `Airborne_SW_Benefits_Of_The_Ada_Language` or a simplified version such as `Airborne_SW_Benefits_Ada_Language`.